utils/scripts/install-centos8-java.sh

```diff
@@ -0,0 +1,14 @@
+#!/bin/sh -e
+
+# For fedora, java-11 is installed along with maven if we install maven from
+# repo. But we need java-8 (1.8). The 'devel' package also needs to be
+# installed specifically.
+# For centos, we can install both java-8 and maven from repo.
+
+if [ -e /etc/fedora-release ]; then
+	dnf install java-1.8.0-openjdk java-1.8.0-openjdk-devel maven-openjdk8
+else
+        dnf install java-1.8.0-openjdk maven
+fi
```

Comment:
I think at this point you can just put it in the initial install-centos8.sh, and not have the second script.
```suggestion
if [ -e /etc/fedora-release ]; then
	dnf install java-1.8.0-openjdk java-1.8.0-openjdk-devel maven-openjdk8
else
        dnf install java-1.8.0-openjdk maven
fi
```

Reply:
It's not only for fedora, but also conditional Java build. check line 247 in dockerfile.centos.8.  

"RUN if [[ "$DAOS_JAVA_BUILD" == "yes" ]]; then "

I need to introduce a parameter to install-centos8.sh to check if it's java build.

Reply:
In all other cases we take the approach that we install all needed components/rpms at the beginning, regardless of if they'll be used or not.  As discussed adding additional parameters will break the docker caching of images and cause the dependencies to be rebuilt unnecessarily, so IMO we should install these packages unconditionally earlier in the process.

What you say does make sense that we should not install them until we need them, and then only if requested however we also build daos and the java client for every PR, and that means loading cached docker images from master that avoids downloading updates however with this code as it is then each PR would need to download the maven packages twice (once for gcc, once for clang) for each distro, and this introduces additional network load, but crucially an additional point of failure.

Reply:
make sense. Let me change it.